source-dropbox/main.go

```diff
@@ -0,0 +1,220 @@
+package main
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox/files"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	log "github.com/sirupsen/logrus"
+)
+
+type config struct {
+	Credentials credentials    `json:"credentials"`
+	Path        string         `json:"path"`
+	Parser      *parser.Config `json:"parser"`
+	MatchKeys   string         `json:"matchKeys,omitempty"`
+	Advanced    advancedConfig `json:"advanced"`
+}
+type advancedConfig struct {
+	AscendingKeys bool `json:"ascendingKeys,omitempty"`
+}
+
+type credentials struct {
+	ApiToken string `json:"apiToken"`
+}
+
+func (c config) Validate() error {
+	requiredProperties := [][]string{
+		{"AzureTenantID", c.Credentials.ApiToken},
+	}
+	for _, req := range requiredProperties {
+		if req[1] == "" {
+			return fmt.Errorf("missing '%s'", req[0])
+		}
+	}
+	return nil
+}
+
+func (c config) DiscoverRoot() string {
+	return c.Path
+}
+
+func (c config) RecommendedName() string {
+	return c.Path
+}
+
+func (c config) FilesAreMonotonic() bool {
+	return c.Advanced.AscendingKeys
+}
+
+func (c config) ParserConfig() *parser.Config {
+	return c.Parser
+}
+
+func (c config) PathRegex() string {
+	return c.MatchKeys
+}
+
+func newDropboxStore(ctx context.Context, cfg config) (*dropboxStore, error) {
+	httpClient, err := GetOAuth2HTTPClient(ctx, dropboxConfig)
+	if err != nil {
+		return &dropboxStore{}, err
+	}
+	config := dropbox.Config{
+		Client:   httpClient,
+		LogLevel: dropbox.LogInfo, // if needed, set the desired logging level. Default is off
```

Comment:
We should probably set this log level to debug, since we typically want user-facing logs at the info level to be carefully crafted so they don't cause confusion.